### ISSUE TITLE
Docparams does not raise `differing-type-doc` for Sphinx documentation

### ISSUE THREAD

**Issue by saroad2** [PyCQA/pylint#6478]
### Bug description

When trying to run *Pylint* on a function with wrong parameter type in docstring, written in *Sphinx* format, no `differing-type-doc` (code W9018) is risen.

For example, the following code doesn't raise no error:
```python
"""This is module documentation."""


def my_func(some_int: int) -> float:
    """
    Some function.

    :param some_int: This is a parameter
    :type some_int: str
    :return: some float
    :rtype: float
    """
    return some_int + 0.5
```

### Configuration

```ini
No configuration
```

### Command used

```shell
pylint --load-plugins=pylint.extensions.docparams a.py
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

`differing-type-doc` should be raised on parameter `some_int`, because it is an `int` variable that was documented as `str`.

### Pylint version

```shell
pylint 2.13.7
astroid 2.11.3
Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
```

### OS / Environment

Windows 10

### Additional dependencies

ansicon==1.89.0
astroid==2.11.3
atomicwrites==1.4.0
attrs==21.2.0
bandit==1.7.2
black==22.1.0
blessed==1.19.1
certifi==2021.10.8
charset-normalizer==2.0.12
click==8.0.1
click-params==0.2.0
colorama==0.4.4
cycler==0.10.0
decorator==5.1.1
dill==0.3.4
eddington==0.0.23
et-xmlfile==1.1.0
Flask==2.0.3
future==0.18.2
gitdb==4.0.9
GitPython==3.1.26
idna==3.3
iniconfig==1.1.1
inquirer==2.9.1
isort==5.10.1
itsdangerous==2.1.0
Jinja2==3.0.3
jinxed==1.1.0
joblib==1.1.0
kiwisolver==1.3.1
lazy-object-proxy==1.7.1
MarkupSafe==2.1.0
matplotlib==3.4.2
mccabe==0.6.1
mypy-extensions==0.4.3
numpy==1.19.3
openpyxl==3.0.9
packaging==21.3
pandas==1.4.1
pathspec==0.9.0
pbr==5.8.1
Pillow==8.2.0
platformdirs==2.4.0
pluggy==1.0.0
prettytable==3.2.0
prompt-tool-kit==1.0.14
py==1.11.0
Pygments==2.11.2
pylint==2.13.7
pyparsing==2.4.7
pytest==6.2.5
python-dateutil==2.8.1
python-editor==1.0.4
pytz==2021.3
PyYAML==6.0
readchar==3.0.5
regex==2022.3.15
requests==2.27.1
scikit-learn==1.0
scipy==1.8.0
seaborn==0.11.2
six==1.16.0
smmap==5.0.0
stevedore==3.5.0
threadpoolctl==3.0.0
toml==0.10.2
tomli==1.2.3
tomli_w==1.0.0
tqdm==4.62.3
typing_extensions==4.0.1
uncertainties==3.1.6
urllib3==1.26.8
validators==0.18.2
wcwidth==0.2.5
websockets==10.2
Werkzeug==2.0.3
wrapt==1.13.3

**Comment by Matesanz** [comment]
Same problem in `pylint == 3.2.7`
